pilot/cmd/pilot-agent/main.go

```diff
@@ -562,6 +566,15 @@ func fromJSON(j string) *meshconfig.RemoteService {
 	return &m
 }
 
+func appendTLSCerts(rs *meshconfig.RemoteService) {
+	if rs.TlsSettings == nil {
```

Comment:
This should be an error

Reply:
I did not understand why it should be an error? `TlsSettings` is optional right?

Reply:
Ah, it is to append if required